### PR TITLE
Support subdirs for cache folder

### PR DESCRIPTION
**Pull request by surfdude75**
Actually it works but do not save cache files if a fs cache folder is set and the image is in a subdir.

### CODE REVIEW

File: lib/storage/fs/index.js [hunk 1]
```diff
@@ -34,6 +34,24 @@ p.fetch = function(options, originalPath, stepsHash, cb) {
   });
 };
 
+function checkDir(filename,cb) {
```
Comment (asilvas) [inline lib/storage/fs/index.js]: `, cb`

File: lib/storage/fs/index.js [hunk 2]
```diff
@@ -34,6 +34,24 @@ p.fetch = function(options, originalPath, stepsHash, cb) {
...
+  var folder = path.dirname(filename);
+  fs.stat(folder, function(err,data) {
```
Comment (asilvas) [inline lib/storage/fs/index.js]: spacing `, data)`

File: lib/storage/fs/index.js [hunk 3]
```diff
@@ -34,6 +34,24 @@ p.fetch = function(options, originalPath, stepsHash, cb) {
...
+    if(err == null) {
```
Comment (asilvas) [inline lib/storage/fs/index.js]: `!err` or `if(!err) return cb();`

File: lib/storage/fs/index.js [hunk 4]
```diff
@@ -34,6 +34,24 @@ p.fetch = function(options, originalPath, stepsHash, cb) {
...
+      cb();
+    } else if(err.code == 'ENOENT') {
+      fs.mkdirp(folder,function(err){
```
Comment (asilvas) [inline lib/storage/fs/index.js]: spacing `, function(err) {`

File: lib/storage/fs/index.js [hunk 5]
```diff
@@ -34,6 +34,24 @@ p.fetch = function(options, originalPath, stepsHash, cb) {
   });
 };
 
+function checkDir(filename,cb) {
+  var folder = path.dirname(filename);
+  fs.stat(folder, function(err,data) {
+    if(err == null) {
+      cb();
+    } else if(err.code == 'ENOENT') {
```
Comment (asilvas) [inline lib/storage/fs/index.js]: `===`

**Comment by asilvas** [comment]
Please update with latest from master.

**Comment by surfdude75** [comment]
Hi,

Just updated. Sorry.

**Comment by asilvas** [comment]
@surfdude75 no longer want this merged? Was just waiting on the minor changes I requested.

**Comment by surfdude75** [comment]
I put this minor changes in another pull request and closed this one.

I thought it would be easier to you review this way.

**Comment by surfdude75** [comment]
Because the master I used was very old.

> Em 17 de out de 2016, à(s) 7:58 PM, Rafael Sobral <rafaelsobral3@gmail.com> escreveu:
> 
> I put this minor changes in another pull request and closed this one.
> 
> I thought it would be easier to you review this way.
> 
> 
>

**Pull request closed without merging**